homeassistant/components/poolsense/binary_sensor.py

```diff
@@ -0,0 +1,78 @@
+"""Support for PoolSense binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+)
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+
+from . import PoolSenseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BINARY_SENSORS = {
+    "pH Status": {
+        "unit": None,
+        "icon": "mdi:pool",
+        "name": "pH Status",
+        "device_class": DEVICE_CLASS_PROBLEM,
+    },
+    "Chlorine Status": {
+        "unit": "",
+        "icon": "mdi:pool",
+        "name": "Chlorine Status",
+        "device_class": DEVICE_CLASS_PROBLEM,
+    },
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    binary_sensors_list = []
+    for binary_sensor in BINARY_SENSORS:
+        binary_sensors_list.append(
+            PoolSenseBinarySensor(
+                coordinator,
+                config_entry.data[CONF_EMAIL],
+                config_entry.data[CONF_PASSWORD],
+                binary_sensor,
+            )
+        )
+
+    async_add_entities(binary_sensors_list, False)
+
+
+class PoolSenseBinarySensor(PoolSenseEntity, BinarySensorEntity):
+    """Representation of PoolSense binary sensors."""
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return self.coordinator.data[self.info_type] == "green"
```

Comment:
Im not too familiar with DEVICE_CLASS_PROBLEM but it lofically seems it would be true if red rather than green unless theres some inverted logic somewhere

Reply:
I have changed this. You are right. I wasn't thinking of the device class, I had it as green on , red off.